ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml

```diff
@@ -114,6 +114,15 @@ Control {
         root.selectedAirdropAddress = accountSelector.selectedAirdropAddress
     }
 
+    function setOldSharedAddresses(oldSharedAddresses) {
```

Comment:
I had to add those ugly functions because the component expects to have the old addresses at the start, but that's not possible sadly, because the call is async.

Let me know if there is a better way of handling it so that we can just use the signal from the view directly instead.

Reply:
I don't have a better idea but you could try resolving the needed properties before opening the dialog, eg. in Popups.qml, in `openEditSharedAddressesPopup`:

```qml
    function openEditSharedAddressesPopup(communityId) {
// <--- resolve the properties here, and pass them to the dialog as `selectedSharedAddresses` and `selectedAirdropAddress`
        openPopup(editSharedAddressesPopupComponent, {communityId: communityId, isEditMode: true})
    }
```

Reply:
That wouldn't work either, since it's async and I can't just pass a callback. We have to wait for the properties to be updated and call the signal.

I could use a Connections instead, but it's not particularly clean either